server/routerlicious/packages/services-client/src/auth.ts

```diff
@@ -32,7 +32,7 @@ export function validateTokenClaims(
 		);
 	}
 
-	if (claims.scopes === undefined || claims.scopes.length === 0) {
```

Comment:
How hard is it to fix the typing of `ITokenClaims`? That would make future issues like this impossible.

Reply:
I do not believe add typing would fix issue here.

Typing restrictions is simply a compile time check, and the `null` get from runtime and from a 3rd party library, type restrictions would not change the runtime behavior if my understanding is correct.

Reply:
I mean to say, currently the definition of `ITokenClaims.scope` is typed as `string[]` or `string[] | undefined`. This typing doesn't accurately reflect reality, so the condition above didn't throw a compile-time error, even though the value could potentially be `null`. If the typing were changed to `string[] | null | undefined`, that would prevent a similar error from slipping through.

Reply:
Oh I see what you mean, good point.